src/tensorwaves/optimizer/callbacks.py

```diff
@@ -30,6 +31,50 @@ def finalize(self) -> None:
         self.__progress_bar.close()
 
 
+class YAMLSummary:
+    def __init__(
+        self,
+        filename: str,
+        step_size: int = 10,
+    ) -> None:
+        """Log fit parameters and the estimator value to a `tf.summary`.
+
+        The logs can be viewed with `TensorBoard
+        <https://www.tensorflow.org/tensorboard>`_ via:
+
+        .. code-block:: bash
+
+            tensorboard --logdir logs
+        """
+        self.__iteration = 0
+        self.__step_size = step_size
+        self.__stream = open(filename, "w")
+
+    def __call__(self, parameters: dict, estimator_value: float) -> None:
+        self.__iteration += 1
+        if self.__iteration % self.__step_size != 0:
+            return
+        output_dict = {
+            "Time": datetime.now(),
+            "Iteration": self.__iteration,
+            "EstimatorValue": float(estimator_value),
+            "Parameters": {
+                name: float(value) for name, value in parameters.items()
+            },
+        }
+        _empty_file(self.__stream)
```

Comment:
we dont want to kept appending?

Reply:
No, that would conflict with YAML syntax

Reply:
Not if everything is wrapped into an array right? But ok that can be a future improvement

Reply:
Yeah but then YAML is perhaps not the appropriate format (rather csv or so)

Reply:
Just another idea. Maybe Yaml is also not the best choice here. And the structure. If we would append there is like a 70% overhead in terms of storage, since the parameter names would be written to disk every time. Some table structure with a header would be more useful.